client/js/lounge.js

```diff
@@ -742,13 +749,29 @@ $(function() {
 		}
 	}
 
-	sidebar.on("click", ".chan, button", function() {
+	sidebar.on("click", ".chan, button", function(e, data) {
 		var self = $(this);
 		var target = self.data("target");
 		if (!target) {
 			return;
 		}
 
+		if (!data || data.pushState !== false) {
+			var state = {
```

Comment:
If target is channel then store channelId, otherwise just the target, no need to hardcode a switch.